assembl/views/api2/attachments.py

```diff
@@ -44,17 +76,42 @@ def get_file(request):
     ctx = request.context
     document = ctx._instance
     f = File.get(document.id)
+    if f.infected:
+        raise HTTPNotAcceptable("Infected with a virus")
     escaped_double_quotes_filename = (f.title
         .replace(u'"', u'\\"')
         .encode('iso-8859-1', 'replace'))
     url_quoted_utf8_filename = url_quote(f.title.encode('utf-8'))
-    return Response(
-        body=f.safe_data(),
+    handoff_to_nginx = asbool(config.get('handoff_to_nginx', False))
+    if handoff_to_nginx:
+        kwargs = dict(body='')
+    else:
+        if 'Range' in request.headers:
+            raise HTTPRequestRangeNotSatisfiable()
+        fs = open(f.path, 'rb')
+        app_iter = None
+        environ = request.environ
+        if 'wsgi.file_wrapper' in environ:
+            app_iter = environ['wsgi.file_wrapper'](fs, _BLOCK_SIZE)
+        if app_iter is None:
+            app_iter = FileIter(fs, _BLOCK_SIZE)
+        kwargs=dict(app_iter=app_iter)
+
+    r = Response(
+        content_length=f.size,
         content_type=str(f.mime_type),
+        last_modified=f.creation_date,
+        expires=datetime.now()+timedelta(days=365),
```

Comment:
This might become an issue in case of deletes/moderation/etc, no?

Reply:
yes, does not play well with deletes, but it's what gives us caching. The user would have to know the file ID...
That's a judgement call. I noticed that you never edit files, but replace them, so we don't worry about that. I am going this way, you may go another.